change flex version to 3.12

reidliu41 opened the pull request:
<!--
(Thanks for sending a pull request!
--> 
1.  From the docker hub, it showed flex and flex-3.12 using the same docker file.
https://hub.docker.com/r/openemr/openemr/

2.  But from the /root/devtools, it seems that there is no "-p" for the PHP command, it cannot see the progress without the "-p" option. Or there are more different, so suggest to change flex version to 3.12 or edge by default.
```
[root@easy-env openemr]# docker ps
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS              PORTS                                         NAMES
9c542742b66a        openemr/openemr:flex-3.12   "./run_openemr.sh"       41 minutes ago      Up 41 minutes       0.0.0.0:8300->80/tcp, 0.0.0.0:9300->443/tcp   openemr_openemr_1
0f20753258ba        phpmyadmin/phpmyadmin       "/docker-entrypoint.…"   41 minutes ago      Up 41 minutes       0.0.0.0:8310->80/tcp                          openemr_phpmyadmin_1
7cb5477673e0        mariadb:10.5                "docker-entrypoint.s…"   41 minutes ago      Up 41 minutes       0.0.0.0:8320->3306/tcp                        openemr_mysql_1
[root@easy-env openemr]# openemr-cmd -d openemr_openemr_1 e "grep memory_limit /root/devtools"
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcs -p -n --extensions=php,inc --report-width=120 --standard=ci/phpcs.xml --report=full .
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcs -p -n --extensions=php,inc --report-width=120 --standard=ci/phpcs_src.xml --report=full src/
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcbf -p -n --extensions=php,inc --report-width=120 --standard=ci/phpcs.xml --report=full .
[root@easy-env openemr]# openemr-cmd s
/var/www/localhost/htdocs # /root/devtools psr12-report
Generating PSR12 code styling error report
............................................................   60 / 2818 (2%)    <<<< progress
............................................................  120 / 2818 (4%)
..........................................................^C


[root@easy-env openemr]# docker ps
CONTAINER ID        IMAGE                   COMMAND                  CREATED             STATUS              PORTS                                         NAMES
4c5c3c83fd24        openemr/openemr:flex    "./run_openemr.sh"       14 minutes ago      Up 14 minutes       0.0.0.0:8300->80/tcp, 0.0.0.0:9300->443/tcp   openemr_openemr_1
1cf07155dd38        phpmyadmin/phpmyadmin   "/docker-entrypoint.…"   14 minutes ago      Up 14 minutes       0.0.0.0:8310->80/tcp                          openemr_phpmyadmin_1
99da7c94e983        mariadb:10.5            "docker-entrypoint.s…"   14 minutes ago      Up 14 minutes       0.0.0.0:8320->3306/tcp                        openemr_mysql_1
[root@easy-env openemr]# openemr-cmd -d openemr_openemr_1 e "grep memory_limit /root/devtools"
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcs -n --extensions=php,inc --standard=ci/phpcs.xml --report=full .     <<<<
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcs -n --extensions=php,inc --standard=ci/phpcs_src.xml --report=full src/     <<<
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcbf -n --extensions=php,inc --standard=ci/phpcs.xml .  <<<<
[root@easy-env openemr]# openemr-cmd s
/var/www/localhost/htdocs # /root/devtools  psr12-report
Generating PSR12 code styling error report          << without progress
```

<!-- Add the issue number that is fixed by this PR (In the form Fixes #123) -->
Fixes #

#### Short description of what this resolves:


#### Changes proposed in this pull request:

-
-
-

bradymiller commented:
hi @reidliu41 ,

Do you have most recent flex docker? Need to do :
```
docker pull openemr/openemr:flex
```

This is a common issue (folks not realizing that need to manually grab most recent docker). Pinning the version would make this more complicated since then devs would need to pull flex-3.12 which is further complicated by this changing every 4-6 months when an new alpine version is released.

bradymiller commented:
If still seeing issue even after the pull, then what do you see for them when do:
```
docker images
```

reidliu41 commented:
@bradymiller, ok. got it. 
1. I just git pull the code to my dev env, and the run docker-compose to run, and flex is set in docker-compose.yml by default.
2. And when I test "psr12-report" or "..fix", it really strange that there is no output, finally, I found that is caused by "-p" parameter. Without the output, I think it is not easy to judge run or not. And I checked the DevOps repo, which is only for flex-3.12 and flex-edge. So I suggested. ^^
3. When I tried to pull it manually, it doesn't work, but I remove the flex images, it works. It's a little strange.
```
# docker images
REPOSITORY                                      TAG                 IMAGE ID            CREATED             SIZE
openemr/openemr                                 flex-edge           8554e079ab3a        24 hours ago        685MB
openemr/openemr                                 flex-3.12           860da2f17007        27 hours ago        724MB
mariadb                                         10.5                7c1c380b0dd8        3 weeks ago         407MB
openemr/openemr                                 flex                310c704e5305        5 weeks ago         719MB.  <<< should be the first time

# docker pull openemr/openemr:flex
flex: Pulling from openemr/openemr
Digest: sha256:125f70c126b723b8dfd7f0a154f4bf678e6be89abf76efa1c5eec592ada28933
Status: Downloaded newer image for openemr/openemr:flex
docker.io/openemr/openemr:flex

# docker rmi 310c704e5305
Untagged: openemr/openemr@sha256:b1e94c8e81ed14b288571af375892c80309e37e3b8e5359869f787f2873221f9
Deleted: sha256:310c704e530541baad7117a4f92efc27694fec4f17a2b86eb1761d2b4f184ae6
Deleted: sha256:e4f711060883d3b024a54743c0276574b5359990e2ddd079fe45f69f56711617
Deleted: sha256:365e2bb67ac1d3166255fe49baeab9481239dcbd702a33ae024bc4ec0c80f9b0

# docker images
REPOSITORY                                      TAG                 IMAGE ID            CREATED             SIZE
openemr/openemr                                 flex-edge           8554e079ab3a        24 hours ago        685MB
openemr/openemr                                 flex                860da2f17007        27 hours ago        724MB

# docker pull openemr/openemr:flex
  1 # Use admin/pass as user/password credentials to login to openemr (from OE_USER and OE_PASS below)
flex: Pulling from openemr/openemr
Digest: sha256:125f70c126b723b8dfd7f0a154f4bf678e6be89abf76efa1c5eec592ada28933
Status: Image is up to date for openemr/openemr:flex     <<<<<<<
docker.io/openemr/openemr:flex

# docker ps
CONTAINER ID        IMAGE                   COMMAND                  CREATED             STATUS              PORTS                                                                NAMES
25f83fbcbe58        openemr/openemr:flex    "./run_openemr.sh"       16 minutes ago      Up 16 minutes       0.0.0.0:8300->80/tcp, 0.0.0.0:9300->443/tcp                          openemr_openemr_1

#  openemr-cmd -d openemr_openemr_1 e "grep memory_limit /root/devtools"
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcs -p -n --extensions=php,inc --report-width=120 --standard=ci/phpcs.xml --report=full .
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcs -p -n --extensions=php,inc --report-width=120 --standard=ci/phpcs_src.xml --report=full src/
    php -d memory_limit=640M /root/.composer/vendor/squizlabs/php_codesniffer/bin/phpcbf -p -n --extensions=php,inc --report-width=120 --standard=ci/phpcs.xml --report=full .
```
4. As I understand, it should pull the most recent image for the first time.
